README.md

```diff
@@ -25,3 +25,56 @@ sudo chmod 666 /dev/ttyUSB0
 sudo chown $USER /dev/ttyUSB0
 sudo adduser $USER dialout
 ```
+# System Design
+
+## Modules
+Communication between the Hardware and Game Event Loop modules is done via message queue.
+Hardware events are placed in the queue by the Hardware module then the Game is ticked, this causes the Game to process the events, using the Player Engine to update the state of the game and enqueues Game messages into a queue for the Hardware system to process (resulting in animations to confirm actions were taken).
+The Game Event Loop uses functions in the player engine to update the state of the game.
+```mermaid
+graph
+    A[Hardware] --> B[(Game Message Queue)]
```

Comment:
Are some of these arrows pointing in the wrong direction?

Reply:
They might be. The point of this was just to show the different modules and what their interfaces to each other are.